File: easy/0896_monotonic_array/solution.py
from typing import List


class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True

        increasing = False
        decreasing = False

        for i in range(1, len(nums)):
            if (increasing and nums[i] < nums[i - 1]) or (
                decreasing and nums[i] > nums[i - 1]
            ):
                return False
            elif nums[i] > nums[i - 1]:
                increasing = True
            elif nums[i] < nums[i - 1]:
                decreasing = True

        return True

File: easy/0896_monotonic_array/solution.cpp
using namespace std;

#include <vector>

class Solution {
public:
  bool isMonotonic(vector<int> &nums) {
    const int numsLen = nums.size();

    if (numsLen == 1)
      return true;

    bool increasing = true;
    bool decreasing = true;

    for (int i = 1; i < numsLen; ++i) {
      if (nums[i] < nums[i - 1])
        increasing = false;
      if (nums[i] > nums[i - 1])
        decreasing = false;

      if (!increasing && !decreasing)
        return false;
    }

    return true;
  }
};
